util/urlutil/redact_test.go

```diff
@@ -0,0 +1,45 @@
+package urlutil
+
+import "testing"
+
+func TestRedactCredentials(t *testing.T) {
+	cases := []struct {
+		name string
+		url  string
+		want string
+	}{
+		{
+			name: "non-blank Password",
+			url:  "https://user:password@host.tld/this:that",
+			want: "https://xxxxx:xxxxx@host.tld/this:that",
+		},
+		{
+			name: "blank Password",
+			url:  "https://user@host.tld/this:that",
+			want: "https://xxxxx@host.tld/this:that",
+		},
+		{
+			name: "blank Username",
+			url:  "https://:password@host.tld/this:that",
+			want: "https://:xxxxx@host.tld/this:that",
+		},
+		{
+			name: "blank Username, blank Password",
+			url:  "https://host.tld/this:that",
+			want: "https://host.tld/this:that",
+		},
+		{
+			name: "invalid URL",
+			url:  "1https://foo.com",
+			want: "1https://foo.com",
+		},
+	}
+	for _, tt := range cases {
+		t := t
```

Comment:
Should this be `tt := tt` ?

Reply:
This is actually not necessary, I just removed it.

Reply:
Have you double checked if it's not needed (also when run with `t.Parallel()` / `--parallel` ?)

Reply:
Ok yeah with `t := t` we want the test state, not the test case (`tt`), so as `func(t *testing.T)` runs in a separate goroutine we need it as we use `--parallel`.